### PR TITLE
Cleanup SetWindowPos interop and structify WINDOWPOS/HDLAYOUT

### PR DESCRIPTION
**Pull request by hughbe**
Bad rebase -- doesn't compile. Not yet ready to review !

## Proposed Changes
- Extract common SetWindowPos interop
- Follow common interop guidelines
- Structify HDLAYOUT

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/1914)

### CODE REVIEW

File: src/Common/src/Interop/User32/Interop.SetWindowPos.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        public static IntPtr HWND_TOP = (IntPtr)0;
+        public static IntPtr HWND_BOTTOM = (IntPtr)1;
+        public static IntPtr HWND_TOPMOST = (IntPtr)(-1);
+        public static IntPtr HWND_NOTOPMOST = (IntPtr)(-2);
+        public static IntPtr HWND_MESSAGE = (IntPtr)(-3);
+
+        [DllImport(Libraries.User32, ExactSpelling = true)]
+        public static extern BOOL SetWindowPos(
+            IntPtr hWnd,
+            IntPtr hWndInsertAfter,
```
Comment (zsd4yr) [inline src/Common/src/Interop/User32/Interop.SetWindowPos.cs]: Do we want to do any work to validate hWndInsertAfter https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setwindowpos

Looks like it should be +1 -2 of hWnd

Comment (zsd4yr) [reply src/Common/src/Interop/User32/Interop.SetWindowPos.cs]: Maybe make those an enum?

Comment (hughbe) [reply src/Common/src/Interop/User32/Interop.SetWindowPos.cs]: C# doesn’t support IntPtr enums sadly

File: src/System.Windows.Forms/src/System/Windows/Forms/CommonDialog.cs [hunk 2]
```diff
@@ -94,8 +94,14 @@ private protected static void MoveToScreenCenter(IntPtr hWnd)
             Rectangle screen = Screen.GetWorkingArea(Control.MousePosition);
             int x = screen.X + (screen.Width - r.right + r.left) / 2;
             int y = screen.Y + (screen.Height - r.bottom + r.top) / 3;
-            SafeNativeMethods.SetWindowPos(new HandleRef(null, hWnd), NativeMethods.NullHandleRef, x, y, 0, 0, NativeMethods.SWP_NOSIZE |
-                                 NativeMethods.SWP_NOZORDER | NativeMethods.SWP_NOACTIVATE);
+            User32.SetWindowPos(
```
Comment (zsd4yr) [inline src/System.Windows.Forms/src/System/Windows/Forms/CommonDialog.cs]:
```c#
User32.SetWindowPos(
    hWnd,
    User32.HWND_TOP,
    x,
    y,
    flags: User32.SWP.NOSIZE | User32.SWP.NOZORDER | User32.SWP.NOACTIVATE);
```

**Pull request merged**